projecteuler/problems/finds.py

```diff
@@ -0,0 +1,39 @@
+#!/usr/bin/env python3
+
+from projecteuler.helpers import primenumbers, factors, number_checks
+
+def find_largest_prime_factor(number):
+    primes = primenumbers.get_prime_numbers(0, number)
+    if not primes:
+        return None
+    
+    all_factors = factors.get_factors(number)
+    if not all_factors:
+        return None
+
+    prime_factors = set(primes).intersection(all_factors)
+    return None if not prime_factors else max(prime_factors)
+
+def find_largest_palindromic_sum(min_number_to_sum, max_number_to_sum):
+    for max_number in range(max_number_to_sum, 0, -1):
+        for min_number in range(max_number_to_sum, min_number_to_sum, -1):
+            sum_number = max_number * min_number
+            if number_checks.is_palindromic_number(sum_number):
+                return sum_number
+    return None
+
+def find_smallest_multiple(min_number, max_number):
+    if min_number <= 0 or max_number <= 0:
+        raise TypeError("Min and max inputs must be positive whole numbers")
+    
+    count=1
+    while(True):
+        for divisible_number in range(min_number, max_number+1):
```

Comment:
the `for/else` construct is frowned upon for being confusing, despite it being a great idea. The `else` clause is executed if the entire for loop ran to completion, which is the logical opposite of an `if/else` where the else only runs if the `if` block  didn't run at all. Arguably it'd be better to call it `for/then`.

To simplify this bit, check out the `any` and `all` built-in functions combined with generator expressions.

Reply:
Ah cool, used all now with a generator (hadn't looked at generator vs list before)